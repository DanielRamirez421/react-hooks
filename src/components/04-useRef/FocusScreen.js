import React, { useRef } from 'react';

import './../03-examples/MultipleCustomHooks.css'

export const FocusScreen = () => {

  const inputRef = useRef();

  console.log(inputRef);

  const handleClick = () => {
    // document.querySelector('input').select();
    inputRef.current.select();
    console.log(inputRef);
  }

  return (
  <div>
    <h3>FocusScreen!</h3>
    <hr />

    <input 
      ref={ inputRef }
      className='form-control'
      placeholder='your name'/>

    <button 
      className='btn btn-outline-primary mt-5'
      onClick={ handleClick }>
      Focus
    </button>
  </div>
  );
};
