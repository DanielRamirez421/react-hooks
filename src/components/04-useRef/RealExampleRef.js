import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

import '../03-examples/MultipleCustomHooks.css';

export const RealExampleRef = () => {

  const [show, setShow] = useState(false);

  return (
  <div>
    <h1>RealExampleRef!</h1>
    <hr />

    <button 
      className='btn btn-outline-primary mb-3' 
      onClick={ () => setShow(!show) }
      >Toggle
      </button>
    { show && <MultipleCustomHooks /> }
  </div>
  );
};
