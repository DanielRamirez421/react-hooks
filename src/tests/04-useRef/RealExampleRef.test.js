
import { shallow } from "enzyme";
import { RealExampleRef } from "../../components/04-useRef/RealExampleRef";

describe('RealExampleRef component tests', () => { 
  
  const wrapper = shallow( <RealExampleRef /> );

  test('should show component', () => { 
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBeFalsy();
  });

  test('should show component <MultipleCustomHook />', () => { 
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBeTruthy();
  });  

 });