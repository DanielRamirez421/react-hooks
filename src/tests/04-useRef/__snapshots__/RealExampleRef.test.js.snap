// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef component tests should show component 1`] = `
<div>
  <h1>
    RealExampleRef!
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mb-3"
    onClick={[Function]}
  >
    Toggle
  </button>
</div>
`;
